### PR TITLE
dynamicModelId of the Automaton model must be unique in groovy file

### PR DESCRIPTION
**Pull request by thangqp**
dynamicModelId of the Automaton model must be unique (not the name of model) in groovy file => must be generated from model name and equipment id from automaton

### CODE REVIEW

File: src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.gridsuite.mapping.server.utils;
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java:1]: License header

File: src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+import org.gridsuite.mapping.server.dto.automata.AbstractAutomaton;
+
+public interface IAutomatonIdProvider {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java:5]: authorship

Comment (geofjamg) [inline src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java:5]: We usually do not use these kind of convention naming for interface like I.... We prefer AutomatonIdProvider -> AutomatonIdProviderImpl

File: src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    default String getId(AbstractAutomaton automaton) {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java:6]: default should only be used for backward compatibility but not for implementing a method.

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java (hunk 3 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java:6]: You are right!

File: src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java (hunk 2 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java:5]: OK

**Review by gautierbureau — APPROVED** [review]

File: src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java (hunk 1 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java]: Added

File: src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java (hunk 2 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/mapping/server/utils/IAutomatonIdProvider.java]: Added

**Review by geofjamg — APPROVED** [review]

**Pull request merged**